src/connection.cpp

```diff
@@ -488,12 +487,16 @@ namespace oxen::quic
                 break;
             }
 
-            log::info(log_cat, "Sending data packet with non-stream data frames");
-            if (auto rv = send_packet(nwrite); rv != 0)
-                return;
-            ngtcp2_conn_update_pkt_tx_time(conn.get(), ts);
+            buf.second = nwrite;
+
+            if (++n_packets == batch_size)
+                break;
```

Comment:
This doesn't seem right: this `for(;;)` loop may want to send more than 1 packet, but this loop could be restricted to just one iteration (if the previous loop filled up 7/8 slots).  I think we need the same append-and-send-if-full logic as in the upper loop.